src/psscale.c

```diff
@@ -1002,6 +1002,21 @@ GMT_LOCAL void psscale_draw_colorbar (struct GMT_CTRL *GMT, struct PSSCALE_CTRL
 		}
 	}
 
+	/* Scale parameters to sqrt of the ratio of color bar length to default map dimension of 15 cm */
+	scale_down = sqrt (MAX (Ctrl->D.dim[GMT_X], Ctrl->D.dim[GMT_Y]) / (15.0 / 2.54));
+	GMT->current.setting.font_annot[GMT_PRIMARY].size *= scale_down;
+	GMT->current.setting.font_annot[GMT_SECONDARY].size *= scale_down;
+	GMT->current.setting.font_label.size *= scale_down;
+	GMT->current.setting.map_frame_pen.width *= scale_down;
+	GMT->current.setting.map_tick_pen[GMT_PRIMARY].width *= scale_down;
+	GMT->current.setting.map_tick_pen[GMT_SECONDARY].width *= scale_down;
+	GMT->current.setting.map_tick_length[GMT_ANNOT_UPPER] *= scale_down;
+	GMT->current.setting.map_tick_length[GMT_TICK_UPPER] *= scale_down;
+	GMT->current.setting.map_annot_offset[GMT_PRIMARY] *= scale_down;
+	GMT->current.setting.map_annot_offset[GMT_SECONDARY] *= scale_down;
+	GMT->current.setting.map_label_offset[GMT_X] *= scale_down;
+	GMT->current.setting.map_label_offset[GMT_Y] *= scale_down;
+	fprintf (stderr, "scale-down = %lg\n", scale_down);
```

Comment:
Should remove this `fprintf` or change it to GMT DEBUG MESSAGE:
```suggestion
	fprintf (stderr, "scale-down = %lg\n", scale_down);
```